.travis.yml

```diff
@@ -8,8 +8,27 @@ perl:
   - "5.14"
   - "5.12"
   - "5.10"
+matrix:
+  include:
+    - perl: 5.22
+      env: COVERAGE=1   # enables coverage+coveralls reporting
+      include:
+    - perl: 5.24
+      env: COVERAGE=1   # enables coverage+coveralls reporting
+sudo: false
 before_install:
   - git clone git://github.com/travis-perl/helpers ~/travis-perl-helpers
   - source ~/travis-perl-helpers/init
   - build-perl
   - perl -V
+  - build-dist
+  - cd $BUILD_DIR             # $BUILD_DIR is set by the build-dist command
+install:
+  - cpan-install ExtUtils::MakeMaker~6.68 --deps
+  - cpan-install --coverage
+before_script:
+  -  coverage-setup
+script:
+  - prove -l -j$(test-jobs) $(test-files)   # parallel testing
+after_success:
+  - coverage-report
```

Comment:
It looks to me like most of these are just the default `perl-helpers` behaviour.  Have you tried seeing what happens if we delete everything that aligns with the defaults and try:

```
before_install:
  - eval $(curl https://travis-perl.github.io/init) --auto
```

?

See https://github.com/travis-perl/helpers